core/src/main/kotlin/xtdb/trie/DataRel.kt

```diff
@@ -39,14 +41,24 @@ interface DataRel<L> : AutoCloseable {
 
     class Arrow(
         private val al: BufferAllocator, private val bp: BufferPool,
-        private val dataFile: Path, override val schema: Schema
+        private val dataFile: Path, override val schema: Schema, private val level: Long
     ) : DataRel<ArrowHashTrie.Leaf> {
 
         private val relsToClose = mutableListOf<Relation>()
 
         override fun loadPage(leaf: ArrowHashTrie.Leaf): RelationReader =
             bp.getRecordBatch(dataFile, leaf.dataPageIndex).use { rb ->
-                Relation.fromRecordBatch(al, schema, rb).also { relsToClose.add(it) }
+                Relation.fromRecordBatch(al, schema, rb).let { standardRel ->
+                    if (level == 0L) {
+                        resolveSameSystemTimeEvents(al, standardRel).also {
+                            standardRel.close()
+                            relsToClose.add(it)
+                        }
```

Comment:
@mbutlerw This extra bitemporal resolution is actually only needed when compacting and not when scanning. I am now wondering if it should be skipped in scanning.

Reply:
unless this would ease the complexity in the code you've already written, I wouldn't worry about this - in practice, we're not going to be scanning from L0s for long before someone compacts them into L1

Reply:
Just checked this assumption again. `MergePlanPage` is used in scan.clj instead of `DataRel` so actually not issue for now. Maybe something we can come back to once we have completely switched to the new `xtdb.arrow` implementation and the merging happens the same way in both scanning and compacting.